numba_cuda/numba/cuda/tests/cudapy/test_inspect.py

```diff
@@ -37,28 +47,47 @@ def foo(x, y):
         # Signature in annotation
         self.assertIn("(float32, int32)", typeanno)
         file.close()
-        # Function name in LLVM
-        llvm = foo.inspect_llvm(sig)
-        self.assertIn("foo", llvm)
 
-        # Kernel in LLVM
-        self.assertIn("define void @", llvm)
-
-        asm = foo.inspect_asm(sig)
-
-        # Function name in PTX
-        self.assertIn("foo", asm)
-        # NVVM inserted comments in PTX
-        self.assertIn("Generated by NVIDIA NVVM Compiler", asm)
+        filecheck = FileCheckKernel(foo)
```

Comment:
I think `foo` is decorated with cuda.jit and is returned as a `CUDADispatcher` object. You might not access the `__doc__ ` attribute from original kernel.